src/main/kotlin/io/github/yuutoproject/yuutobot/commands/Dialog.kt

```diff
@@ -18,54 +18,153 @@
 
 package io.github.yuutoproject.yuutobot.commands
 
+import com.fasterxml.jackson.core.type.TypeReference
 import io.github.yuutoproject.yuutobot.commands.base.AbstractCommand
 import io.github.yuutoproject.yuutobot.commands.base.CommandCategory
+import io.github.yuutoproject.yuutobot.utils.EMOJI_REGEX
+import io.github.yuutoproject.yuutobot.utils.NONASCII_REGEX
+import io.github.yuutoproject.yuutobot.utils.httpClient
+import io.github.yuutoproject.yuutobot.utils.jackson
 import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
+import net.dv8tion.jda.api.utils.data.DataObject
 import net.dv8tion.jda.internal.utils.IOUtil
 import okhttp3.Call
 import okhttp3.MediaType.Companion.toMediaType
-import okhttp3.OkHttpClient
 import okhttp3.Request
 import okhttp3.RequestBody.Companion.toRequestBody
 import okhttp3.Response
+import org.slf4j.LoggerFactory
 import java.io.IOException
 import okhttp3.Callback as OkHttp3Callback
 
-// TODO: get arch to implement the actual command
-class Dialog : AbstractCommand("dialog", CommandCategory.INFO, "does something", "[bg] <char> <text>") {
+class Dialog : AbstractCommand("dialog", CommandCategory.INFO, "Generates an image of a character in Camp Buddy saying anything you want", "[bg] <char> <text>") {
+    private val logger = LoggerFactory.getLogger(this.javaClass)
+
+    private val backgrounds: List<String>
+    private val characters: List<String>
+
+    init {
+        val bgAndChars = getBackgroundsAndCharacters()
+
+        backgrounds = bgAndChars.first
+        characters = bgAndChars.second
```

Comment:
Remove the function and put the logic here; it's only used once.

Reply:
I disagree with that, putting the logic there would clutter the init block and that is not what you want

Reply:
It is the purpose of the init block to initialize the class, and in this case getting the backgrounds and characters is fundamental to the operation of the class. What clutter? This is literally the only logic in the block right now.

Reply:
That is not true, the init block gets called after the class is already initialized (after the constructor), the only reason to use the init block here is so we don't get a "bgAndChars" getter in the bytecode.

Reply:
Also if you look at the code in the method you can see that it is quite complex and does stuff like making HTTP requests that do not belong in an init block

Reply:
There's no reason to create a function for a single use when you can put that logic in the relevant section. Especially since the init block is empty otherwise. It would be clear what it's doing from the fact that the code is assigning to backgrounds and characters. If that is not sufficient, that's what comments are for - not functions.